src/formatDate.js

```diff
@@ -48,9 +48,67 @@
  *
  * @returns {string}
  */
+const DAY_SIGN = 'DD';
+const MONTH_SIGN = 'MM';
+const YEAR_SIGN = 'YYYY';
+const YEAR_SIGN_SHORT = 'YY';
+const CENTURY_19 = '19';
+const CENTURY_20 = '20';
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldFormatDate = date.split(fromFormat[3]);
+  const newFormatDate = [];
+  let newYearLength = 0;
+  let oldYearLength = 0;
+  let year,
+    month,
+    day,
+    yearIndex;
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i].includes(YEAR_SIGN_SHORT)) {
+      year = oldFormatDate[i];
+      oldYearLength = fromFormat[i].length;
+    }
+
+    if (fromFormat[i].includes(MONTH_SIGN)) {
+      month = oldFormatDate[i];
+    }
+
+    if (fromFormat[i].includes(DAY_SIGN)) {
+      day = oldFormatDate[i];
+    }
+  }
+
+  for (const part of toFormat) {
+    if (part.includes(YEAR_SIGN_SHORT)) {
+      newYearLength = part.length;
+    }
+  }
+
+  if (newYearLength === 2) {
+    yearIndex = toFormat.indexOf(YEAR_SIGN_SHORT);
+    newFormatDate[yearIndex] = year.slice(2);
+  } else {
+    yearIndex = toFormat.indexOf(YEAR_SIGN);
+    newFormatDate[yearIndex] = year;
+  }
+
+  if (oldYearLength === 2 && newYearLength === 4) {
+    newFormatDate[yearIndex] = CENTURY_20 + year;
+
+    if (year >= 30) {
+      newFormatDate[yearIndex] = CENTURY_19 + year;
+    }
+  }
+
+  const monthIndex = toFormat.indexOf(MONTH_SIGN);
+  const dayIndex = toFormat.indexOf(DAY_SIGN);
+
+  newFormatDate[monthIndex] = month;
+  newFormatDate[dayIndex] = day;
+
+  return newFormatDate.join(toFormat[3]);
```

Comment:
For better readability of the code, we can separate individual variables at the beginning of the function:
```js
const oldSeparator = fromFormat[3];
const newSeparator = toFormat[3];
```